packages/blocks/src/types.ts

```diff
@@ -0,0 +1,1504 @@
+
+	/**
+	 * By default, all blocks will appear in the Gutenberg inserter. To
+	 * hide a block so that it can only be inserted programmatically, set
+	 * to false
+	 *
+	 * (default) true
+	 */
+	inserter?: boolean;
+
+	/**
+	 * A non-multiple block can be inserted into each post, one time only.
+	 *
+	 * (default) true
+	 */
+	multiple?: boolean;
+
+	/**
+	 * By default all blocks can be converted to a reusable block.
+	 *
+	 * (default) true
+	 */
+	reusable?: boolean;
+
+	/**
+	 * This value signals that a block supports some of the CSS style
+	 * properties related to spacing.
+	 *
+	 * When the block declares support for a specific spacing property,
+	 * the attributes definition is extended to include the `style` attribute.
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#spacing BlockSupports.spacing} on developer.wordpress.org
+	 */
+	spacing?: Partial< SpacingProps >;
+
+	/**
+	 * A block may want to disable the ability to toggle the lock state.
+	 * It can be locked/unlocked by a user from the block “Options”
+	 * dropdown by default. To disable this behavior, set `lock` to `false`.
+	 *
+	 * (default) true
+	 */
+	lock?: boolean;
+
+	/**
+	 * A block may want to disable the ability to toggle the lock state.
+	 * It can be locked/unlocked by a user from the block “Options”
+	 * dropdown by default. To disable this behavior, set `lock` to `false`.
+	 */
+	typography?: Partial< TypographyProps >;
+
+	/**
+	 * TODO Undocumented
+	 *
+	 * Used in [private-selector.js](./store/private-selector.js)
+	 */
+	shadow: any;
+
+	/**
+	 * This value signals that a block supports some of the CSS style properties related to
+	 * dimensions.
+	 *
+	 * (default) null
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#dimensions BlockSupports.dimensions} on developer.wordpress.org
+	 */
+	dimensions: BlockDimensions;
+
+	/**
+	 * This value signals that a block supports some of the CSS style properties related to position.
+	 *
+	 * (default) null
+	 *
+	 * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#position BlockSupports.position} on developer.wordpress.org
+	 */
+	position: BlockPosition;
+
+	/**
+	 * TODO Undocumented
+	 *
+	 * @internal
+	 */
+	__experimentalSelector: any;
+}
+
+//
+// Attributes
+// ----------------------------------------------------------------------------
+
+/**
+ * Types of data stored by block attributes.
+ *
+ * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/#type-validation Type Validation} on developer.wordpress.org
+ */
+export type AttributeSchemaType =
+	| 'string'
+	| 'boolean'
+	| 'object'
+	| 'null'
+	| 'array'
+	| 'integer'
+	| 'number';
+
+export type BaseAttributeSource = {
+	type: AttributeSchemaType | AttributeSchemaType[];
+};
+
+/**
+ * @public
+ */
+export namespace AttributeSource {
```

Comment:
I thought we were wanting to avoid namespaces in TypeScript…

Reply:
Sure thing, that came over from the DefinitelyTyped declaration. I had added AttributeSource to the beginning of everything and it looked pretty verbose, then I moved them to a module, because that is the preferred way to do a namespace, but I wasn't sure what would be preferred so I left it as is.

Reply:
What would you suggest?

Reply:
I will do that. I was debating on breaking this file up and organizing the types into separate files in a `types` folder. Does that sound like a good idea?

Reply:
seems like it. can't hurt to try. the blocks package is quite large

Reply:
I broke everything into separate files. Removes the "namespace", but technically a module namespace now.